packages/indexer-agent/src/agent.ts

```diff
@@ -50,6 +53,48 @@ const uniqueDeployments = (
   deployments: SubgraphDeploymentID[],
 ): SubgraphDeploymentID[] => deployments.filter(uniqueDeploymentsOnly)
 
+const convertSubgraphBasedRulesToDeploymentBased = (
+  rules: IndexingRuleAttributes[],
+  subgraphs: Subgraph[],
+  previousVersionBuffer: number,
+): IndexingRuleAttributes[] => {
+  const toAdd: IndexingRuleAttributes[] = []
+  rules.map(rule => {
+    const ruleSubgraph = subgraphs.find(
+      subgraph => subgraph.id == rule.identifier,
+    )
+    if (ruleSubgraph) {
+      const latestVersion = ruleSubgraph.versionCount
```

Comment:
a bit nit-picking: it might be intuitive if we let `const latestVersion = ruleSubgraph.versionCount - 1` so that below you can find `latestDeploymentVersion` directly with `latestVersion`, and `previousDeploymentVersion` with `latestVersion - 1`?